src/main.rs

```diff
@@ -1,92 +1,77 @@
-use futures::{
-    channel::mpsc::{channel, Receiver},
-    SinkExt, StreamExt,
-};
 use notify::{Event, RecommendedWatcher, RecursiveMode, Watcher};
-use std::{
-    panic,
-    path::{Path, PathBuf},
-};
-use std::{thread, time};
+use std::path::Path;
+use std::time;
+use tokio::sync::mpsc::{channel, Receiver};
 
 mod chmod;
 mod permissions;
 
 fn async_watcher() -> notify::Result<(RecommendedWatcher, Receiver<notify::Result<Event>>)> {
-    let (mut tx, rx) = channel(1);
+    let (tx, rx) = channel(512);
 
     // Automatically select the best implementation for your platform.
     // You can also access each implementation directly e.g. INotifyWatcher.
     let watcher = RecommendedWatcher::new(move |res| {
-        futures::executor::block_on(async {
-            match tx.send(res).await {
-                Ok(()) => {}
-                Err(err) => println!("watch error: {:?}", err),
-            };
-        })
+        match tx.blocking_send(res) {
```

Comment:
With tokio we can do a blocking_send directly, not exactly sure how it works under the hood but I think it doesn't block (and doesn't require a thread spawn).